### PR TITLE
#54o data sort by project date

### PR DESCRIPTION
**Pull request by alesyalis**

### CODE REVIEW

File: Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs [hunk 1]
```diff
@@ -10,6 +10,13 @@ public void Apply(ODataModelBuilder builder, ApiVersion apiVersion, string route
         {
             builder.EntitySet<ProjectDTO>("Projects");
             builder.ComplexType<ProjectPrimarySkillDTO>();
+           /* builder.EntityType<ProjectDTO>().Collection
+                 .Function("GetAllProject")
+                 .ReturnsCollectionFromEntitySet<ProjectDTO>("Projects");*/
+            builder.EntitySet<ShortProjectDTO>("AllProjects");
+            builder.EntityType<ShortProjectDTO>().Collection
+                  .Function("GetAllProject")
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs:18]: GetShortProjects

File: Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs [hunk 2]
```diff
@@ -10,6 +10,13 @@ public void Apply(ODataModelBuilder builder, ApiVersion apiVersion, string route
         {
             builder.EntitySet<ProjectDTO>("Projects");
             builder.ComplexType<ProjectPrimarySkillDTO>();
+           /* builder.EntityType<ProjectDTO>().Collection
+                 .Function("GetAllProject")
+                 .ReturnsCollectionFromEntitySet<ProjectDTO>("Projects");*/
+            builder.EntitySet<ShortProjectDTO>("AllProjects");
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs:16]: ShortProjects

File: Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs [hunk 3]
```diff
@@ -28,5 +29,17 @@ public IQueryable<ProjectDTO> Get()
...
+
+        [EnableQuery(
+            HandleNullPropagation = HandleNullPropagationOption.False,
+            AllowedQueryOptions = AllowedQueryOptions.Filter | AllowedQueryOptions.OrderBy | AllowedQueryOptions.Top | AllowedQueryOptions.Skip | AllowedQueryOptions.Count)]
+        public IEnumerable<ShortProjectDTO> GetAllProject(ODataQueryOptions<ShortProjectDTO> options)
+        {
+            var candidates = _projectService.Get();
+            var filteredCandidates = options.ApplyTo(candidates) as IEnumerable<ShortProjectDTO>;
+            var result = filteredCandidates!.Where(x => x.IsActive = true).Where(x => x.StartDate >= DateTime.Now)
+                                            .OrderBy(x => x.StartDate);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs:41]: StartRegistrationDate everywhere

File: Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs (hunk 1 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs:18]: done

File: Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs (hunk 2 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs]: done

File: Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs (hunk 3 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs:41]: done

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:3]: ?

File: Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs (hunk 3 above)
Comment (elizavetapavliv) [reply Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs:40]: StartRegistrationDate

File: Recrutify/Recrutify.Services/Services/ProjectService.cs [hunk 5]
```diff
@@ -53,6 +53,12 @@ public async Task<List<ProjectDTO>> GetAllAsync()
             return _mapper.Map<List<ProjectDTO>>(projects);
         }
 
+        public IQueryable<ShortProjectDTO> GetShort()
+        {
+            var projects = _projectRepository.Get();
+            return _mapper.Map<IQueryable<ShortProjectDTO>>(projects);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/ProjectService.cs:59]: no, see how mapping performed in another method

Comment (alesyalis) [reply Recrutify/Recrutify.Services/Services/ProjectService.cs]: done

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs (hunk 4 above)
Comment (alesyalis) [reply Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs]: done

File: Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs (hunk 3 above)
Comment (elizavetapavliv) [reply Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs:43]: IsActive==true

Comment (elizavetapavliv) [reply Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs:43]: Only one Where should be with conditions via &&

File: Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs [hunk 6]
```diff
@@ -8,8 +8,12 @@ public class ProjectsConfiguration : IModelConfiguration
     {
         public void Apply(ODataModelBuilder builder, ApiVersion apiVersion, string routePrefix)
         {
-            builder.EntitySet<ProjectDTO>("Projects");
+            builder.EntityType<ProjectDTO>();
             builder.ComplexType<ProjectPrimarySkillDTO>();
+            builder.EntitySet<ShortProjectDTO>("Projects");
+            builder.EntityType<ShortProjectDTO>().Collection
+                  .Function("GetShortProjects")
+                  .ReturnsCollectionFromEntitySet<ShortProjectDTO>("ShortProjects");
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs:16]: Try "Projects"

File: Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs (hunk 3 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs:43]: done

Comment (elizavetapavliv) [reply Recrutify/Recrutify.Host/Controllers/OData/ProjectsController.cs:43]: Orderby StartRegistrationDate

File: Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs (hunk 6 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/OData/ProjectsConfiguration.cs]: done

**Review by elizavetapavliv — APPROVED** [review]

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**